### ISSUE TITLE
Support for cython pure python syntax: cython.declare

### ISSUE THREAD

**Issue by jtoledo1974** [pylint-dev/pylint#8024]
### Bug description

"""Test for cython pure python cython.declare"""
import cython

MY_VAR = cython.declare(cython.int, 0)

### Configuration

```ini
[testoptions]
requires: cython
```

### Command used

```shell
pylint regression_cython_declare.py
```

### Pylint output

```shell
************* Module regression_cython_declare
tests/functional/r/regression_02/regression_cython_declare.py:4:24: E1101: Module 'cython' has no 'int' member; maybe 'bint'? (no-member)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

------------------------------------
Your code has been rated at 10.00/10

### Pylint version

```shell
pylint 2.16.0-dev
astroid 2.12.13
Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]
```

### OS / Environment

Needs cython. The default cython package as of today is
Cython==0.29.32

**Comment by jtoledo1974** [comment]
Cython has a "pure python" mode with which you can write your python code using valid python syntax.

I believe pylint should be able to recognize it and avoid reporting correct syntax when using cython. In this case, it is expected that the cython module doesn't provide the members. 

I'll do a pull request with the test case, and will try to produce a patch.

**Comment by jtoledo1974** [comment]
For reference, and if I fail to produce a patch, a workaround is to add this to the pyproject.toml file:

```toml
[tool.pylint.BASIC]
generated-members = "cython.*"
```

**Comment by Pierre-Sassoulas** [comment]
Isn't a problem since [cython/cython@6e31b41f6](https://github.com/cython/cython/commit/6e31b41f6353bd623d98aa4de40d3645965e47ba), but ``cython.declare(cython.int, 0)`` still raises E1101 on main with Cython 0.29.33 through 3.2.9 on latest pylint main.
